airflow/providers/google/cloud/transfers/bigquery_to_sql.py

```diff
@@ -0,0 +1,126 @@
+    """
+    Fetches the data from a BigQuery table (alternatively fetch data for selected columns)
+    and insert that data into a SQL table.
+
+    .. note::
+        If you pass fields to ``selected_fields`` which are in different order than the
+        order of columns already in
+        BQ table, the data will still be in the order of BQ table.
+        For example if the BQ table has 3 columns as
+        ``[A,B,C]`` and you pass 'B,A' in the ``selected_fields``
+        the data would still be of the form ``'A,B'`` and passed through this form
+        to the SQL database.
+
+    :param dataset_table: A dotted ``<dataset>.<table>``: the big query table of origin
+    :param selected_fields: List of fields to return (comma-separated). If
+        unspecified, all fields are returned.
+    :param gcp_conn_id: reference to a specific Google Cloud hook.
+    :param database: name of database which overwrite defined one in connection
+    :param replace: Whether to replace instead of insert
+    :param batch_size: The number of rows to take in each batch
+    :param location: The location used for the operation.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields: Sequence[str] = (
+        "sql_table",
+        "impersonation_chain",
+    )
+
+    def __init__(
+        self,
+        *,
+        dataset_table: str,
+        sql_table: str,
+        selected_fields: list[str] | str | None = None,
+        gcp_conn_id: str = "google_cloud_default",
+        database: str | None = None,
+        replace: bool = False,
+        batch_size: int = 1000,
+        location: str | None = None,
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.selected_fields = selected_fields
+        self.gcp_conn_id = gcp_conn_id
+        self.database = database
+        self.sql_table = sql_table
+        self.replace = replace
+        self.batch_size = batch_size
+        self.location = location
+        self.impersonation_chain = impersonation_chain
+        try:
+            self.dataset_id, self.table_id = dataset_table.split(".")
+        except ValueError:
+            raise ValueError(f"Could not parse {dataset_table} as <dataset>.<table>") from None
+
+    @abc.abstractmethod
+    def get_sql_hook(self) -> DbApiHook:
+        pass
```

Comment:
I understand your opinion that it might seem redundant, however:
1. In my opinion - it's better to get used to writing docstrings starting with self-explanatory methods, as it will become necessary for more complicated functions. That being said, for simple methods - one line could definitely do.
2. Putting opinions aside, it's required by [PEP 257](https://peps.python.org/pep-0257/), at least for public methods :)

Reply:
> Public methods (including the __init__ constructor) should also have docstrings.

The wording of PEP 257 suggests it's optional.

In Clean Code, Robert Martin enumerates a collection of smells and heuristics including:

> C3: Redundant Comment
> 
> A comment is redundant if it describes something that adequately describes itself. For example:
> i++; // increment i
> 
> Another example is a Javadoc that says nothing more than (or even less than) the function signature.
> 
> Comments should say things that the code cannot say for itself.

As suggested, I put my opinion aside and described `persist_links` with `"""This function persists the connection to the SQL provider"""`.